cloud/openstack/os_server.py

```diff
@@ -360,13 +361,18 @@ def _create_server(module, cloud):
             metas[k] = v
         module.params['meta'] = metas
 
+    if type(module.params['security_groups']) is str:
+        module.params['security_groups'] = module.params['security_groups'].split(',')
```

Comment:
Instead of this,  it would be better to set `type='list'` in `openstack_full_argument_spec`.  `type='list'` automatically handles this functionality.